ts/src/htx.ts

```diff
@@ -5065,6 +5106,135 @@ export default class htx extends Exchange {
+        let market = undefined;
+        let marginMode = undefined;
+        for (let i = 0; i < orders.length; i++) {
+            const rawOrder = orders[i];
+            const marketId = this.safeString (rawOrder, 'symbol');
+            if (symbol === undefined) {
+                symbol = marketId;
+            } else {
+                if (symbol !== marketId) {
+                    throw new BadRequest (this.id + ' createOrders() requires all orders to have the same symbol');
+                }
+            }
+            const type = this.safeString (rawOrder, 'type');
+            const side = this.safeString (rawOrder, 'side');
+            const amount = this.safeValue (rawOrder, 'amount');
+            const price = this.safeValue (rawOrder, 'price');
+            const orderParams = this.safeValue (rawOrder, 'params', {});
+            const marginResult = this.handleMarginModeAndParams ('createOrders', orderParams);
+            const currentMarginMode = marginResult[0];
+            if (currentMarginMode !== undefined) {
+                if (marginMode === undefined) {
+                    marginMode = currentMarginMode;
+                } else {
+                    if (marginMode !== currentMarginMode) {
+                        throw new BadRequest (this.id + ' createOrders() requires all orders to have the same margin mode (isolated or cross)');
+                    }
+                }
+            }
+            market = this.market (symbol);
+            let orderRequest = undefined;
+            if (market['spot']) {
+                orderRequest = await this.createSpotOrderRequest (marketId, type, side, amount, price, orderParams);
+            } else {
+                orderRequest = this.createContractOrderRequest (marketId, type, side, amount, price, orderParams);
+            }
+            orderRequest = this.omit (orderRequest, 'marginMode');
+            ordersRequests.push (orderRequest);
+        }
+        const request = {};
+        let response = undefined;
+        if (market['spot']) {
+            response = await this.privatePostOrderBatchOrders (ordersRequests);
+        } else {
+            request['orders_data'] = ordersRequests;
+            if (market['linear']) {
+                marginMode = (marginMode === undefined) ? 'cross' : marginMode;
+                if (marginMode === 'isolated') {
+                    response = await this.contractPrivatePostLinearSwapApiV1SwapBatchorder (request);
+                } else if (marginMode === 'cross') {
+                    response = await this.contractPrivatePostLinearSwapApiV1SwapCrossBatchorder (request);
+                }
+            } else if (market['inverse']) {
+                if (market['swap']) {
+                    response = await this.contractPrivatePostSwapApiV1SwapBatchorder (request);
+                } else if (market['future']) {
+                    response = await this.contractPrivatePostV1ContractBatchorder (request);
```

Comment:
can't find this endpoint 
![image](https://github.com/ccxt/ccxt/assets/43336371/c47f9d34-eb03-45a8-84c7-64b4815933fd)